src/types/algokit-core-bridge.ts

```diff
@@ -1,23 +1,37 @@
 import * as algodApi from '@algorandfoundation/algokit-algod-api'
-import { addressFromString, Transaction as AlgokitCoreTransaction, encodeTransactionRaw } from '@algorandfoundation/algokit-transact'
+import {
+  addressFromString,
+  Transaction as AlgokitCoreTransaction,
+  assignFee,
+  encodeTransactionRaw,
+} from '@algorandfoundation/algokit-transact'
 import algosdk, { Address, TokenHeader } from 'algosdk'
+import { CommonTransactionParams } from './composer'
 
 function getAlgokitCoreAddress(address: string | Address) {
   return addressFromString(typeof address === 'string' ? address : address.toString())
 }
 
 // Experimental feature to build algosdk payment transactions with algokit-core
-export function buildPayment({
-  sender,
-  receiver,
-  amount,
-  closeRemainderTo,
-  rekeyTo,
-  note,
-  lease,
-  suggestedParams,
-}: algosdk.PaymentTransactionParams & algosdk.CommonTransactionParams) {
-  const txnModel: AlgokitCoreTransaction = {
+export function buildPayment(
+  params: CommonTransactionParams,
+  {
+    sender,
+    receiver,
+    amount,
+    closeRemainderTo,
+    rekeyTo,
+    note,
+    lease,
+    suggestedParams,
+  }: algosdk.PaymentTransactionParams & algosdk.CommonTransactionParams,
+) {
+  if (params.staticFee !== undefined) {
```

Comment:
If we wanted to we could do away with using flatFee and changing the meaning of fee.